runtime/src/accounts_background_service.rs

```diff
@@ -385,26 +385,38 @@ impl SnapshotRequestHandler {
         let mut snapshot_time = Measure::start("snapshot_time");
         let snapshot_storages = snapshot_utils::get_snapshot_storages(&snapshot_root_bank);
         let accounts_package = match request_type {
-            SnapshotRequestType::Snapshot => {
-                let bank_snapshot_info = snapshot_utils::add_bank_snapshot(
-                    &self.snapshot_config.bank_snapshots_dir,
-                    &snapshot_root_bank,
-                    &snapshot_storages,
-                    self.snapshot_config.snapshot_version,
-                    status_cache_slot_deltas,
-                )?;
-                AccountsPackage::new_for_snapshot(
-                    accounts_package_type,
-                    &snapshot_root_bank,
-                    &bank_snapshot_info,
-                    &self.snapshot_config.full_snapshot_archives_dir,
-                    &self.snapshot_config.incremental_snapshot_archives_dir,
-                    snapshot_storages,
-                    self.snapshot_config.archive_format,
-                    self.snapshot_config.snapshot_version,
-                    accounts_hash_for_testing,
-                )
-            }
+            SnapshotRequestType::Snapshot => match &accounts_package_type {
+                AccountsPackageType::Snapshot(_) => {
+                    let bank_snapshot_info = snapshot_utils::add_bank_snapshot(
```

Comment:
I have a concern here. Are we subtlely breaking something else that expects the bank snapshot to be there even for the `AccountsHashVerifier` packages?

My guess is no, but I think I need to look in more detail into how exactly these `AccountPackage` types are used.